src/features/tasklist/components/TeamHeader.tsx

```diff
@@ -0,0 +1,70 @@
+import BgTeamPattern from "@/assets/images/bg-team-pattern.png";
+import Dropdown, { DropdownOption } from "@/components/dropdown";
+import Icon from "@/components/icon";
+import Image from "next/image";
+
+interface TeamHeaderProps {
+  teamName: string;
+  isAdmin?: boolean;
+  onEdit?: () => void;
+  onDelete?: () => void;
+}
+
+export default function TeamHeader({
+  teamName,
+  isAdmin = false,
+  onEdit,
+  onDelete,
+}: TeamHeaderProps) {
+  const options: DropdownOption[] = [
+    { label: "수정하기", value: "edit" },
+    { label: "삭제하기", value: "delete" },
+  ];
+
+  const handleSelect = (option: DropdownOption | string) => {
+    if (typeof option === "string") return;
+    switch (option.value) {
+      case "edit":
+        onEdit?.();
+        break;
+      case "delete":
+        onDelete?.();
+        break;
+    }
+  };
+
+  return (
+    <>
+      <header className="flex h-7 w-full max-w-[1120px] items-center justify-start rounded-xl bg-none shadow-[0px_15px_50px_-12px_rgba(0,0,0,0.05)] outline-border-primary desktop:h-16 desktop:justify-between desktop:bg-background-primary desktop:px-7 desktop:py-4 desktop:outline">
+        <h1 className="text-lg-b text-text-primary tablet:text-2xl-b">
+          {teamName}
+        </h1>
+
+        <Image
+          src={BgTeamPattern}
+          alt=""
+          aria-hidden
+          width={326}
+          height={102}
+          className="mr-2.5 ml-auto hidden desktop:block"
+        />
+
+        {isAdmin && (
+          <Dropdown
+            anchor={
+              <button
+                aria-label="팀 설정 열기"
+                className="ml-2 cursor-pointer py-1"
+              >
+                <Icon name="gear" size="large" />
+              </button>
+            }
+            options={options}
+            alignment="left"
+            onSelect={handleSelect}
```

Comment:
- `options`에 `DropdownOption` type을 전달하고 있어서, `onSelect` callback에는 `string` type이 전달될 수 없습니다.
- 아래와 같이 작성하면 `handleSelect`에서 `string` type check를 따로 하지 않아도 됩니다.
  ```suggestion
              onSelect={(option) => handleSelect(option as DropdownOption)}
  ```

Reply:
이 부분을 계속 헤매다가 일단 에러만 안 나게 처리했었는데
작성해주신 것처럼 하면 되는군요! 감사합니다👍🏻👍🏻